### PR TITLE
fix(linux): async custom protocol crashing sending response

### PR DESCRIPTION
**Pull request by lucasfernog**
Fixes `GTK may only be used from the main thread. thread 'tokio-runtime-worker' panicked at .cargo/registry/src/index.crates.io-6f17d22bba15001f/webkit2gtk-2.0.1/src/auto/uri_scheme_response.rs:26`

Ref https://github.com/tauri-apps/tauri/issues/9794

### CODE REVIEW

File: src/webkitgtk/web_context.rs [hunk 1]
```diff
@@ -361,28 +361,33 @@ where
...
-            .headers()
-            .get(CONTENT_TYPE)
-            .and_then(|h| h.to_str().ok());
-
-          let response = URISchemeResponse::new(&input, buffer.len() as i64);
-          response.set_status(http_response.status().as_u16() as u32, None);
-          if let Some(content_type) = content_type {
-            response.set_content_type(content_type);
-          }
+          gtk::glib::idle_add(move || {
```
Comment (amrbashir) [inline src/webkitgtk/web_context.rs]: should we use [`MainContext::invoke`](https://docs.rs/glib/0.18.5/glib/struct.MainContext.html#method.invoke) instead? seems like `idle_add` might wait until the loop is idle which may impact performance, also `MainContext::invoke` will call the closure directly if it is already on the main thread.

Comment (lucasfernog) [reply src/webkitgtk/web_context.rs]: oh i'll try that

Comment (lucasfernog) [reply src/webkitgtk/web_context.rs]: sorted @amrbashir

Comment (lucasfernog) [reply src/webkitgtk/web_context.rs]: i knew that function existed - just couldn't find it in tao/wry

Comment (amrbashir) [reply src/webkitgtk/web_context.rs]: > sorted @amrbashir

did you forget to push :joy:?

Comment (lucasfernog) [reply src/webkitgtk/web_context.rs]: 🤣🤣🤣🤣 in my defense i was testing this while holding a baby

Comment (amrbashir) [reply src/webkitgtk/web_context.rs]: no worries, anything for the baby ❤️

Comment (lucasfernog) [reply src/webkitgtk/web_context.rs]: finally pushed

**Review by amrbashir — APPROVED** [review]

**Pull request merged**